doc/explanation/package-management.md

```diff
@@ -0,0 +1,226 @@
+package lock.
+
+To do so, Dune uses
+[`opam-0install-solver`](https://github.com/ocaml-opam/opam-0install-solver),
+which is a variant of the `0install` solver to find solutions for opam packages.
+
+Contrary to opam, the Dune solver always starts from a blank slate. It
+assumes nothing is installed and everything needs to be installed. This has the
+advantage that solving is now simpler, and previous solver solutions don't interfere
+with the current one. Thus, given the same inputs, it should always have the same
+outputs; no state is held between the solver runs.
+
+This can lead to more packages being installed (as opam won't install new package versions
+by default if the existing versions satisfy the
+constraints), but it avoids interference from already installed packages that
+lead to potentially different solutions.
+
+After solving is done, the solution gets written into the lock directory with
+all the metadata necessary to build and install the packages. From this
+point on, there is no need to access any package repositories.
+
+:::{note}
+Solving and locking does not download the package sources. These are downloaded
+in a later step.
+:::
+
+## Building
+
+When building, Dune will read the information from the lock directory and set
+up rules to use the packages. Check {doc}`/explanation/mental-model` for
+details about rules.
+
+The rules that the package management sets up include:
+
+* Fetch rules to download the sources as well as any additional sources like
+  patches and unpack them
+* Build rules to evaluate the build instructions from the build instructions
+  stored in the lock directory
+* Install rules to put the artifacts that were built into the appropriate
+  Dune-managed folders
+
+Creating these processes as rules mean that they will only be executed on
+demand, so if the project has already downloaded the sources, it does not
+need to download them again. Likewise, if packages are installed, they stay
+installed.
+
+The results of the rules are stored in the project's `_build` directory
+and managed automatically by Dune. Thus, when cleaning the build directory, the
+installed packages are cleaned as well and will be reinstalled at the next
+build.
+
+When building the users project, the installed packages are added to the
+necessary search paths, so user code can use the dependencies without any
+additional configuration.
+
+(compatibility)=
+## Packaging for Dune Compatibility
+
+Dune can build and install most packages as dependencies, even if they are not
+built with Dune themselves. Dune will execute the build instructions from the
+lock file, very similar to opam.
+
+However, packages must meet certain requirements to be compatible with Dune.
+
+The most important one is that the packages must not use absolute paths to
+refer to files. That means, they cannot read the path they are being built or
+installed into and expect this path to be stable. Dune builds packages in a
```

Comment:
```suggestion
refer to files. This means they cannot rely on the path they are being built or installed in, as it may not remain consistent. Dune builds packages in a
```
Is this still accurate? If so, it improves readability and clarity. If not, let's work on clarifying it together.

Reply:
I've updated it a bit to remove the commas. I don't think "consistent" is the right word in the context as that would imply the path is "inconsistent" but that doesn't make that much sense. I've paraphrased it to be "stay the same" which might be simpler to understand. WDYT?